rmf_site_editor/src/site/scenario.rs

```diff
@@ -102,65 +135,214 @@ pub fn update_current_scenario(
+                        children.iter().for_each(|e| {
+                            subtree_dependents.insert(*e);
+                            queue.push(*e);
+                        });
+                    }
+                }
+                // Only insert new instance in children scenarios. Other parent/root
+                // scenarios will not have access to this instance
+                for dependent in subtree_dependents.drain() {
+                    if let Ok((child_entity, _)) = scenarios.get(dependent) {
+                        commands
+                            .spawn(Instance::new_inherited(None))
+                            .insert(Affiliation(Some(update.instance)))
+                            .set_parent(child_entity);
+                    }
                 }
+            }
+            _ => {
+                let instance_entities =
+                    get_scenario_instance_entities(update.scenario, &children, &scenario_instances);
+                let Some(((mut instance, _), scenario_child)) = instance_entities
+                    .iter()
+                    .find(|(_, i)| *i == update.instance)
+                    .and_then(|(c_entity, _)| {
+                        scenario_instances
+                            .get_mut(*c_entity)
+                            .ok()
+                            .zip(Some(c_entity))
+                    })
+                else {
+                    continue;
+                };
+                let instance = instance.as_mut();
+
+                match update.update_type {
+                    UpdateInstanceType::Include => {
+                        let instance_pose = instance.pose().or(recall_instance
+                            .get(*scenario_child)
+                            .ok()
+                            .and_then(|r| r.pose));
 
-                let existing_moved_instance = current_scenario
-                    .moved_instances
-                    .iter_mut()
-                    .find(|(e, _)| *e == entity);
-                if let Some(existing_moved_instance) = existing_moved_instance {
-                    existing_moved_instance.1 = pose.clone();
-                    return;
-                } else {
-                    current_scenario
-                        .moved_instances
-                        .push((entity, pose.clone()));
-                    return;
+                        if parent_pose.is_some() {
+                            *instance = Instance::new_inherited(instance_pose)
+                        } else if let Some(instance_pose) = instance_pose {
+                            *instance = Instance::new_added(instance_pose);
+                        } else {
+                            let instance_id = model_instances
+                                .get(update.instance)
+                                .map(|n| n.0.clone())
+                                .unwrap_or(format!("{}", update.instance.index()).to_string());
```

Comment:
You should use `.unwrap_or_else(|_| ...)` here, otherwise you're telling the program to format this string every single time.

Also I don't think `.to_string()` is needed since `format!` always outputs a String.